android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -1703,40 +1707,34 @@ public void onLocationError(int errorCode)
   @Override
   public void onLocationNotFound()
   {
-    showLocationNotFoundDialog();
+    if (mLocationErrorDialogAnnoying || (mLocationErrorDialog != null && mLocationErrorDialog.isShowing()))
+      return;
+
+    final String message = String.format("%s\n\n%s", getString(R.string.current_location_unknown_message),
+        getString(R.string.current_location_unknown_title));
+    mLocationErrorDialog = new AlertDialog.Builder(this)
+        .setMessage(message)
+        .setOnDismissListener(dialog -> mLocationErrorDialog = null)
+        .setNegativeButton(R.string.current_location_unknown_stop_button, (dialog, which) ->
+        {
+          LocationHelper.INSTANCE.setStopLocationUpdateByUser(true);
+          LocationHelper.INSTANCE.stop();
```

Comment:
```suggestion
          final LocationHelper lh = LocationHelper.INSTANCE;
          lh.setStopLocationUpdateByUser(true);
          lh.stop();
```

Reply:
Then I wasted my time reviewing this piece of code. Using a bad pattern everywhere is not the best approach.